pip/req/req_set.py

```diff
@@ -570,13 +602,32 @@ def _pip_has_created_build_dir(self):
             )
         )
 
+    def _to_install(self):
+        """Create the installation order.
+
+        We install the user specified things in the order given, except when
+        dependencies require putting other things first.
```

Comment:
hmm,  "nothing to do with" is  a bit extreme.  :  )

you reference #2478 as what you're trying to fix, and that was a user wanting topo sort, because he was working with packages that weren't using `setup_requires` properly.

regardless of the motivation, I think it's a good idea.  can't hurt.